### PR TITLE
Docs: Argument Clinic: Move the CConverter class to the reference

### PR DESCRIPTION
**Pull request by erlend-aasland**
----
:books: Documentation preview :books:: https://cpython-previews--107671.org.readthedocs.build/

### CODE REVIEW

File: Doc/howto/clinic.rst [hunk 1]
```diff
@@ -193,6 +193,71 @@ The CLI supports the following options:
...
 
 
+.. _clinic-classes:
+
+Classes for extending Argument Clinic
+-------------------------------------
+
+.. module:: clinic
+
+.. class:: CConverter
```
Comment (ezio-melotti) [inline Doc/howto/clinic.rst:203]: Is this an actual class that I can import and use?  If not, should it get a `:noindex:`?

I was going through the howto and saw an example that points to [`Modules/zlibmodule.c`](https://github.com/python/cpython/tree/3.11/Modules/zlibmodule.c), but in the source there's no mention of `CConverter`.

**Review by ezio-melotti — APPROVED** [review]

File: Doc/howto/clinic.rst (hunk 1 above)
Comment (erlend-aasland) [reply Doc/howto/clinic.rst:203]: You must subclass the CConverter class when you write custom converters. Perhaps we can point to `Modules/posixmodule.c` instead; there's a lot of custom converters there.

Comment (erlend-aasland) [reply Doc/howto/clinic.rst:203]: Answering your first question; you cannot import it, but you can use it. It is "magically" there when you write a `[python input]` block.

Comment (erlend-aasland) [reply Doc/howto/clinic.rst:203]: I'm not sure about the `:noindex:`, though.

**Pull request merged**